src/blueair_api/device_aws.py

```diff
@@ -12,6 +12,9 @@ class DeviceAws(CallbacksMixin):
     name: str = None
```

Comment:
These attribute declarations typically do not mean anything in Python (some type checkers may use them, but  due to backward compatibility its value is limited): I was mistyping water_shortage to wshortage and no checks were triggered. 

To make typed properties we can declare 
```
@property
def water_shortage(self) -> bool:
  """Returns True if the device reports a water shortage."""
  return safely_get_json_value(self._last_info['state'], ....)
```

That said I don't know whether the current style is conventional in homeassistant codebase. If you have no objections I will update the device_aws.py file to the strongly typed style (would require a update on the ha_blueair custom_component that I will file as well, since the attributes will be made read-only.